### PR TITLE
Verify builds work before PRs are merged

### PR DESCRIPTION
**Pull request by lexicalninja**
## Description of Change

creates processes for #8267 

There are a few things here, we may consider breaking into 2 or 3 different PRs for atomic purposes:

1. adds ios and android bundle to the `code_checks.yml` file. This allows us to make sure any changes to the app arent breaking the js bundling process. 

2. adds a set of build check work for android and ios when we change any files in their native directories. Each one has a new lane added to the fastfile, and three new yml files to make sure each check only counts once. [see this for an explanation ](https://github.com/orgs/community/discussions/36680#discussioncomment-3935835)

## Testing

workflows all pass and only one version runs for each platform.

## Reviewer Validations

code quality, readability, etc
Should this be multiple PRs?

Once this is merged, we will need to test that a PR that does not include files from these directories fires the skip script and does not fire a duplicate action. if that works, these can be made required checks. 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: .github/workflows/native_build_check_android.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    paths:
+      - "VAMobile/android/**"
+
+jobs:
+  android_native_build:
+    name: Android Test Build
+    uses: ./.github/workflows/build_android.yml
+    secrets: inherit
+    with:
+      lane: test_build_only
```
[the remark this answers is not in the document]
Comment (lexicalninja) [reply .github/workflows/native_build_check_android.yml:15]: These two checks are on the actual building of app .ipa or .aab file to make sure there are no issues if we update the native files. updating those files could change the build settings for the final artifacts and cause errors that bundle wont catch

[the remark this answers is not in the document]
Comment (timwright12) [reply .github/workflows/native_build_check_android.yml:15]: yeah I'm hesitant to include fastlane checks in a PR since there are other reasons that can fail including random timeouts.

[the remark this answers is not in the document]
Comment (lexicalninja) [reply .github/workflows/native_build_check_android.yml:15]: these checks do not happen on a regular PR, only when something in the actual native files changes. 

any changes outside of those files just gets the bundle check along with the tests.

[the remark this answers is not in the document]
Comment (lexicalninja) [reply .github/workflows/native_build_check_android.yml:15]: from the attached ticket 

> There are instances where we may want to confirm that changes in a PR didn't break the app build. This would serve as a gate to prevent changes that don't require manual QA from being merged into develop without confirming the build still works. This is mainly a concern when updating the folders for native code, e.g. VAMobile/android, VAMobile/ios.

[the remark this answers is not in the document]
Comment (lexicalninja) [reply .github/workflows/native_build_check_android.yml:15]: > other reasons that can fail including random timeouts.

You have seen timeouts on the actual build step, or on the uploading/verifying upload steps after the build?

[the remark this answers is not in the document]
Comment (timwright12) [reply .github/workflows/native_build_check_android.yml:15]: We see various instances of fastlane failing. Usually on the upload after the build. If we're not actually uploading, I suppose that's fine

[the remark this answers is not in the document]
Comment (lexicalninja) [reply .github/workflows/native_build_check_android.yml:15]: there shouldnt be timouts on builds, and since we are just building the artifact and not uploading you should not see timeouts. 

With these in place you now have unit tests, e2e tests, bundle tests (that test whether the js transpiles from ts correctly), and these other build tests (that make sure that pods/gradle/native changes dont break the final artifacts )

that covers a huge portion of edge cases of broken stuff

**Review by timwright12 — DISMISSED** [review]

File: VAMobile/android/fastlane/Fastfile [hunk 2]
```diff
@@ -113,6 +113,11 @@ platform :android do
     )
   end
 
+  desc "makes a test build to ensure changes to native directories did not brak build process. DOES NOT DISTRIBUTE"
```
Comment (narin) [inline VAMobile/android/fastlane/Fastfile:116]: Typo in "break"

File: .github/workflows/native_build_check_ios_skip.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: '[Code Quality] Check Native iOs Builds'
+#description: 'This test has the opposite filter of native_build_check_ios.yml so that required checks dont hang'
```
Comment (narin) [inline .github/workflows/native_build_check_ios_skip.yml:2]: Are these workflow descriptions meant to be uncommented? If not, please remove `description:` and surrounding apostrophes, and put comment above workflow name to follow patterns of other workflows.

**Review by narin — CHANGES_REQUESTED** [review]
Just some nits but looks good otherwise

**Review by narin — APPROVED** [review]

**Review by IsraelleHub — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]
